rclpy/rclpy/node.py

```diff
@@ -1019,11 +1040,11 @@ def set_parameters_callback(
         """
         Register a set parameters callback.
 
-        Calling this function with override any previously registered callback.
+        Calling this function will override any previously registered callback.
 
         :param callback: The function that is called whenever parameters are set for the node.
         """
-        self._parameters_callback = callback
+        self.add_on_set_paramters_callback(callback)
```

Comment:
Changing the behavior of `set_parameters_callback` seems potentially problematic. Maybe it should be deprecated instead?

Reply:
I agree -- this makes a lot more sense. It's not entirely obvious on first glance [what rclcpp does](https://github.com/ros2/rclcpp/blob/2ed445647400bacbff6420cfb8ccc7bee86ee469/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp#L872-L882) in this case, but it seems like one can set a single callback?

Reply:
I dont think it will cause any problem because it is only used for setting parameter (as of now adding it to the list) and parameters are also exactly the same.
If we modify it we will have to make changes in other files too